src/core/thread/tmf.hpp

```diff
@@ -80,6 +80,15 @@ class Agent : public Coap::Coap
      */
     bool IsTmfMessage(const Ip6::MessageInfo &aMessageInfo) const;
 
+    /**
+     * This method allocates a new MeshCoP message which has Network Control priority level and requires link level
+     * security.
+     *
+     * @returns A pointer to the allocated message, or `nullptr` if allocation fails.
+     *
+     */
+    ot::Coap::Message *NewMeshCoPMessage(void) { return NewPriorityMessage(); }
```

Comment:
Should we move `Coap::NewPriorityMessage()` as `Tmf::Agent::NewMessage()`. We don't need `NewMeshCoPMessage()` and `NewPriorityMessage` in `Coap`.

Reply:
Should we remove `NewPriorityMessage()` from `Coap` class?

Reply:
The `NewPriorityMessage()` method on `Coap` (which is then inherited by `CoapSecure`) is used a bunch of places along with the `Get<Coap::CoapSecure>()`  instance (which is included when we have `OPENTHREAD_CONFIG_DTLS_ENABLE`). For example, in `border_agent.cpp` or in `joiner` (when sending finalize msg). I think it may be useful to keep it.

Reply:
It may be better to use `NewPriorityMessage()` directly everywhere. This actually helps harmonize the way different `MeshCoP` module allocate message.

So closing this PR in favor of 
- https://github.com/openthread/openthread/pull/7274.